core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/kerberos/KeytabProvider.java

```diff
@@ -33,7 +35,12 @@
     public ServiceKeytabResponse getServiceKeytabResponse(Stack stack, GatewayConfig primaryGatewayConfig) {
         ServiceKeytabRequest request = cmServiceKeytabRequestFactory.create(stack, primaryGatewayConfig);
         try {
-            String accountId = ThreadBasedUserCrnProvider.getAccountId();
+            String accountId;
+            if (InternalCrnBuilder.isInternalCrn(ThreadBasedUserCrnProvider.getUserCrn())) {
+                accountId = Crn.fromString(stack.getResourceCrn()).getAccountId();
+            } else {
+                accountId = ThreadBasedUserCrnProvider.getAccountId();
+            }
```

Comment:
please refactor, move the logic to a method, so it would look like:
String accountId = awesomemethod(stack.getreasourcecrn());
You might add some log to the new method so we would see what happened there